ksqldb-parser/src/main/java/io/confluent/ksql/util/ParserUtil.java

```diff
@@ -212,4 +220,31 @@ public static JsonObject convertJsonFieldCase(final JsonObject obj) {
     }
     return new JsonObject(convertedMap);
   }
+
+  /**
+   * Checks if the token is a reserved keyword or not
+   * @param token the String that caused the parsing error
+   * @return true if the token is a reserved keyword according to SqlBase.g4
+   *         false otherwise
+   */
+  public static boolean isReserved(final String token) {
+
+    final SqlBaseLexer sqlBaseLexer = new SqlBaseLexer(
+            new CaseInsensitiveStream(CharStreams.fromString(token)));
+    final CommonTokenStream tokenStream = new CommonTokenStream(sqlBaseLexer);
+    final SqlBaseParser sqlBaseParser = new SqlBaseParser(tokenStream);
+
+    sqlBaseParser.removeErrorListeners();
+
+    final SqlBaseParser.NonReservedContext nonReservedContext = sqlBaseParser.nonReserved();
+    if (nonReservedContext.exception == null) {
```

Comment:
I am a bit confused about the double negation in the comment :P If the token is a reserved token, then there will be an exception thrown?

Reply:
If we call `nonReservedWord` and if it successfully parses, then we just parsed through a `nonReserved` word as defined in `SqlBase.g4`. if it fails (throws an exception) then it still has the potential to be reserved and we continue checking for that.

Reply:
yes, your comment here is better written than the way I phrased it originally! I think Vicky is right perhaps we should have the contrapositive of what I wrote:
```
if no exception was "thrown" when calling the nonReserved rule, then this word matches that rule and is not reserved
```

Reply:
ok done